### PR TITLE
refine data truncate error msg

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
fix https://github.com/pingcap/tidb/issues/7390
In TiDB now:
```SQL
mysql root@127.0.0.1:test> create table t(a bigint);
Query OK, 0 rows affected
Time: 0.008s
mysql root@127.0.0.1:test> insert into t values("asfasdfsajhlkhlksdaf");
(1366, "Incorrect bigint value: 'asfasdfsajhlkhlksdaf' for column 'a' at row 1")
mysql root@127.0.0.1:test> insert into t values("12asdf");
(1366, "Incorrect bigint value: '12asdf' for column 'a' at row 1")
```
In Mysql 
```SQL
mysql root@127.0.0.1:test> create table t (a bigint)
Query OK, 0 rows affected
Time: 0.018s
mysql root@127.0.0.1:test> insert into t values("asfasdfsajhlkhlksdaf");
(1366, "Incorrect integer value: 'asfasdfsajhlkhlksdaf' for column 'a' at row 1")
mysql root@127.0.0.1:test> insert into t values("12asdf");
(1265, "Data truncated for column 'a' at row 1")
```

As you can see, The return error of `insert into t values("12asdf");` in TiDB is still different from MySQL.

### Check List 

Tests

### CODE REVIEW

**Comment by winkyao** [comment]
@crazycs520 Please read the https://github.com/pingcap/tidb/blob/master/CONTRIBUTING.md, and the proper commit message is : 

```
<subsystem>: <what changed>
<BLANK LINE>
<why this change was made>
<BLANK LINE>
<footer>(optional)
```

Please follow this style.

File: table/table.go [hunk 1]
```diff
@@ -77,6 +77,8 @@ var (
 	ErrInvalidRecordKey = terror.ClassTable.New(codeInvalidRecordKey, "invalid record key")
 	// ErrTruncateWrongValue returns for truncate wrong value for field.
 	ErrTruncateWrongValue = terror.ClassTable.New(codeTruncateWrongValue, "incorrect value")
+	// ErrTruncatedWrongValueForField returns for truncate wrong value for field.
+	ErrTruncatedWrongValueForField = terror.ClassTable.New(codeTruncateWrongValue, mysql.MySQLErrName[mysql.ErrTruncatedWrongValueForField])
```
Comment (winkyao) [inline table/table.go]: Add code map in line 213, in tableMySQLErrCodes.

File: executor/executor_test.go [hunk 2]
```diff
@@ -2065,9 +2066,9 @@ func (s *testSuite) TestEmptyEnum(c *C) {
 	tk.MustExec("create table t (e enum('Y', 'N'))")
 	tk.MustExec("set sql_mode='STRICT_TRANS_TABLES'")
 	_, err := tk.Exec("insert into t values (0)")
-	c.Assert(terror.ErrorEqual(err, types.ErrTruncated), IsTrue)
+	c.Assert(terror.ErrorEqual(err, table.ErrTruncatedWrongValueForField), IsTrue)
 	_, err = tk.Exec("insert into t values ('abc')")
-	c.Assert(terror.ErrorEqual(err, types.ErrTruncated), IsTrue)
+	c.Assert(terror.ErrorEqual(err, table.ErrTruncatedWrongValueForField), IsTrue)
```
Comment (winkyao) [inline executor/executor_test.go]: Please make sure the error code is 1366

Comment (crazycs520) [reply executor/executor_test.go]: The error code of `table.ErrTruncatedWrongValueForField` is 1366.

File: executor/insert_common.go [hunk 3]
```diff
@@ -198,11 +201,12 @@ func (e *InsertValues) getRow(cols []*table.Column, list []expression.Expression
 
 	for i, expr := range list {
 		val, err := expr.Eval(chunk.MutRowFromDatums(row).ToRow())
-		if err = e.handleErr(cols[i], rowIdx, err); err != nil {
+		valStr, _ := val.ToString()
```
Comment (shenli) [inline executor/insert_common.go]: Could you move the ToString() operation into the handErr() function? We could use a lazy way to call the ToString function. Or you will call ToString for all the data.

Comment (crazycs520) [reply executor/insert_common.go]: done.

**Comment by crazycs520** [comment]
here is a benchmark about `datum` and `*datum`:
```go
func foo1(d Datum) {
	//	_, _ = d.ToString()
}

func foo2(d *Datum) {
	//	_, _ = d.ToString()
}

func AllDatumSlice() []Datum {
	ds := make([]Datum, 0)
	ds = append(ds, NewIntDatum(1))
	ds = append(ds, NewUintDatum(1))
	ds = append(ds, NewFloat64Datum(28.0))
	ds = append(ds, NewDecimalDatum(NewDecFromStringForTest("72.5")))
	ds = append(ds, NewMysqlBitDatum([]byte{1, 2, 3, 4, 5, 6, 7, 8, 9}))
	ds = append(ds, NewTimeDatum(Time{Time: FromGoTime(time.Now()), Fsp: 6, Type: mysql.TypeTimestamp}))
	return ds
}

func BenchmarkDatum(b *testing.B) {
	ds := AllDatumSlice()
	b.ResetTimer()
	for i := 0; i < b.N; i++ {
		foo1(ds[i%len(ds)])
	}
}

func BenchmarkDatumPoint(b *testing.B) {
	ds := AllDatumSlice()
	b.ResetTimer()
	for i := 0; i < b.N; i++ {
		foo2(&ds[i%len(ds)])
	}
}
```
result is:
```shell
~/code/goread/src/test2 ᐅ go version
go version go1.10.3 darwin/amd64
~/code/goread/src/test2 ᐅ go test -bench=. -v test_test.go
goos: darwin
goarch: amd64
BenchmarkDatum-8        	100000000	        11.5 ns/op
BenchmarkDatumPoint-8   	200000000	         9.37 ns/op
PASS
ok  	command-line-arguments	3.931s
```

**Comment by crazycs520** [comment]
@zz-jason PTAL

**Pull request merged**